client/src/pages/Admin.js

```diff
@@ -182,66 +186,181 @@ FileDownloader.propTypes = {
   text: PropTypes.string.isRequired,
 };
 
-const Configuration = () => {
+const Configuration = ({ lastImport }) => {
+  const [config, setConfig] = useState({
+    roundsPerDuel: 0,
+    questionsPerRound: 0,
+    questionTimerDuration: 0,
+  });
+
+  const [saved, setSaved] = useState(false);
+
+  const updateConfig = (key, value) => {
+    const newKey = { ...config[key], ...{ value } };
+    config[key] = newKey;
+
+    const newConfig = Object.create(null);
+    Object.assign(newConfig, config);
+
+    setConfig(newConfig);
+  };
```

Comment:
I'm not sure to understand what `newKey` is (shouldn't it be `newValue` ?), but the following get around the `Object.assign` method and prevent for overriding with concurrent updates.

```suggestion
  const updateConfig = (key, value) => {
    setConfig((oldConfig => ({ ...oldConfig, [key]:  { ...oldConfig[key], ...{ value } } }));
  };
```